0.11.0 Docs

vector-of-bool opened the pull request:
## This change addresses item #366 

Still a WIP at the moment, but filling out the documentation. Want to make it nice.

I'm adding all as reviewers because I want to make sure I don't have dumb mistakes and be sure that the wording is clear, useful, and unambiguous.

Randshot commented on docs/configuring.rst:
```diff
@@ -117,7 +117,7 @@ using ``${variable}`` syntax. The following built-in variables are expanded:
 
 ``${projectName}``
-    The name of the CMake project. Isn't expanded fully until project has been
+    The name of the CMake project. Isn't expanded fully until project has beenX
```
`beenX` ??

vector-of-bool replied:
Accidental

Randshot commented on docs/kits.rst:
```diff
@@ -3,7 +3,27 @@
+  simply finding the necessary compiler executable. Visual C++ requires certain
+  environment variables to be set to tell it how to find and link to the
+  Visual C++ toolchain headers and libraries.
+- A toolchain file! This is the lowest-level way to instruct CMake how to
```
I think you should stick to surrounding `A toolchain file` with stars `*` and remove the exclamation mark and replace it with `-` just to stay consistent.

Randshot commented on docs/kits.rst:
```diff
@@ -73,13 +128,17 @@ command palette. The following steps are taken to find available kits:
 =========
 
 CMake defines three types of kits: *compiler kits*, *Visual Studio kits*, and
 *toolchain file kits*. They are distinguished by the properties present in
```
I think that `Compiler` and `Toolchain` should be written in upper case.

vector-of-bool replied:
In the phrase `Toolchain Kit` and `Compiler Kit`, probably.

Randshot replied:
yeah

KoeMai commented on docs/kits.rst:
```diff
@@ -12,22 +32,58 @@ CMake Kits
+User-local kits are kits that are available to a particular user for all
+projects open with CMake Tools.
+
+The user-local list of kits is stored in a user-owned file, which you can edit
```
maybe add the path where it is stored, so that the user know that it is stored in the profile folder.

vector-of-bool replied:
Maybe? I'm wary of publicly documenting where this file is stored, as it's path is more of an implementation detail. That's why there is a dedicated "Edit Kits" command: So the user doesn't need to know where it is.

KoeMai replied:
One of the questions on issue reporting was "where is the kit file?". 
But you are right it is acceeable by "edit kit".

Randshot commented on docs/variants.rst:
```diff
@@ -8,8 +8,23 @@ and combine a common set of build options and give them a useful name. The main
 For smaller or simple projects, providing a custom ``cmake-variants.yaml`` is
-unnecessary, and the default CMake build types will work just fine.
+unnecessary, and the default CMake build types will work just fine. Large
+projects with more complex configuration options make want to specify
```
`complex configuration options make want to specify` -> `complex configuration options may want to specify`

Randshot replied:
still relevant

Randshot commented on docs/variants.rst:
```diff
@@ -86,26 +101,138 @@ The option is itself a map with the following keys:
+#. All ``settings`` from the chosen options are passed as ``-D`` arguments to
+   the CMake process.
+#. The ``buildType`` is used for ``CMAKE_BUILD_TYPE``, the ``--config``
+   parameter to the build (For multi-conf generators), and for the CTest
```
`parameter to the build (For multi-conf generators)` -> `parameter for the build (For multi-conf generators)`

KoeMai commented on docs/variants.rst:
```diff
@@ -86,26 +101,138 @@ The option is itself a map with the following keys:
+    A lengthier human-readable string to describe the option.
 
-``buildType``
+``buildType`` (Optional)
```
Is it possible to use `description`, `short`, `long`, `buildtype`, `linkage`, `settings `and `env ` in the setting part, outside of the choices. If not then please describe setting and the allowed choice paramter separate.

A example is shown below:

   ```
 buildType:
      default: debug,
      linkage: shared  #<----
      choices:
        debug:
          short: Debug
          long: Emit debug information
          buildType: Debug
        release:
          short: Release
          long: Optimize generated code
          buildType: Release
        .....
```

KoeMai commented on docs/variants.rst:
```diff
@@ -28,18 +43,18 @@ The variants file can be placed in the root of the project directory, or in the
-variants. When we ask to change the build type, CMake Tools will present each
-possible combination in a quickpick:
+of the possible choices. Two **settings** each with two **options** creates
+*four* variants. When we ask to change the build type, CMake Tools will present
```
I would introduce variant a a important word. This underlines the meaning of the word and the meaning for later sections.

vector-of-bool commented:
I've pushed up the docs to https://vector-of-bool.github.io/docs/vscode-cmake-tools/index.html so that you may view them in their nice rendered output. Almost got them all done.

Randshot commented on docs/building.rst:
```diff
@@ -0,0 +1,104 @@
+Build Flags
+===========
+
+CMake Tools builds using the ``--build`` flag to CMake. This flag is used as
```
```CMake Tools builds using the ``--build`` flag to CMake.```
This doesn't sound right...

Randshot commented on docs/configuring.rst:
```diff
@@ -1,145 +1,166 @@
+How CMake Tools Configures
+**************************
+
+CMake Tools speaks to CMake over *CMake Server*, a execution mode of CMake
```
a execution mode -> an execution mode

Randshot commented on docs/debugging.rst:
```diff
@@ -0,0 +1,113 @@
+
+CMake Tools removes some of the friction required in setting up debugging.
+Because C and C++ projects may define multiple (sometimes dozens or even h
+hundreds) of executables, creating a ``launch.json`` may be difficult, tedious,
```
one `h` too much

Randshot commented on docs/getting_started.rst:
```diff
@@ -77,10 +107,12 @@ command from the command palette. A quick-pick will appear:
-CMake Tools will load up the default variant when it doesn't detect a prior
-selection. For the default variant, this will be ``Debug``.
+.. note::
+    Just like with kits, CMake will ask you which variant to build if you
```
CMake -> CMake Tools

Randshot commented on docs/how_do_i.rst:
```diff
@@ -0,0 +1,59 @@
+******************************************
+
+CMake Tool's won't do this on its own, but you can `use compile_commands.json
+in conjunction with Microsoft's cpptools extension.
```
unnecessary dot `.` here

Randshot commented on docs/settings.rst:
```diff
@@ -0,0 +1,230 @@
+
+If specified, sets a value for ``CMAKE_INSTALL_PREFIX`` when running CMake
+configure. If not, no value will be passed.
+*Note that if ``CMAKE_INSTALL_PREFIX`` is set via ``cmake.configureArgs`` or
```
https://vector-of-bool.github.io/docs/vscode-cmake-tools/settings.html#cmake-installprefix
Isn't displayed correctly :/

Randshot commented on docs/settings.rst:
```diff
@@ -0,0 +1,230 @@
+
+.. _var-subs:
+
+``cmake.configureArgs``
```
Duplicate a couple lines down.

vector-of-bool commented:
I've addressed the commentary above, and all the tests should be running now. I'm feeling ready to merge this, unless anyone has any last-minute objections.
